tensorboard/plugins/pr_curve/tf_pr_curve_dashboard/tf-pr-curve-dashboard.html

```diff
@@ -0,0 +1,256 @@
+  import {getRouter} from '../tf-backend/router.js';
+  import {categorizeTags} from '../tf-categorization-utils/categorizationUtils.js';
+  import {runsColorScale} from "../tf-color-scale/colorScale.js";
+
+  Polymer({
+    is: 'tf-pr-curve-dashboard',
+    properties: {
+      // Determines how the time entry is shown. One of step, relative, or
+      // wall_time.
+      _timeDisplayType : {
+        type: String,
+        value: 'step',
+      },
+      _selectedRuns: Array,
+      _runToTag: Object,  // map<run: string, tags: string[]>
+      // The steps that the step slider for each run should use.
+      _runToAvailableTimeEntries: {
+        type: Object, // map<run: string, steps: number[]>
+        value: {},
+      },
+      // A list of runs that are both selected and have PR curve data.
+      _relevantSelectedRuns: {
+        type: Array,
+        computed: "_computeRelevantSelectedRuns(_selectedRuns, _runToTag)",
+      },
+      _runsWithPrCurveData: Array,  // string[]
+      // The actual step value that each run should use. If a run + tag lacks a
+      // PR curve at this exact step value, the greatest step value less than
+      // this value will be used.
+      _runToStep: {
+        type: Object,  // map<run: string, step: number>
+        notify: true,
+      },
+      _dataNotFound: Boolean,
+      _tagFilter: {
+        type: String,  // upward bound from paper-input
+        value: '.*',
+      },
+      _categories: {
+        type: Array,
+        computed: '_makeCategories(_runToTag, _selectedRuns, _tagFilter)',
+      },
+      _requestManager: {
+        type: Object,
+        value: () => new RequestManager(),
+      },
+      _step: {
+        type: Number,
+        value: 0,
+        notify: true,
+      },
+    },
+    ready() {
+      this.reload();
+    },
+    reload() {
+      Promise.all(
+          [this._fetchTags(), this._fetchTimeEntriesPerRun()]).then(() => {
+        this._reloadCards();
+      });
+    },
+    _fetchTags() {
+      const url = getRouter().pluginRoute('pr_curves', '/tags');
+      return this._requestManager.request(url).then(runToTagToContent => {
+        const runToTag = _.mapValues(runToTagToContent, o => _.keys(o));
+        if (_.isEqual(runToTag, this._runToTag)) {
+          // No need to update anything if there are no changes.
+          return;
+        }
+        const tags = getTags(runToTag);
+        this.set('_dataNotFound', tags.length === 0);
+        this.set('_runToTag', runToTag);
+      });
+    },
+    _fetchTimeEntriesPerRun() {
+      const url = getRouter().pluginRoute(
+          'pr_curves', '/available_time_entries');
+      return this._requestManager.request(url).then(timeEntriesPerRun => {
+        // Compute the relative field for each time entry. For each step, the
+        // relative field is the number of seconds (float) since the first step.
+        _.forOwn(timeEntriesPerRun, (entries, run) => {
+          if (!entries.length) {
```

Comment:
This isn't needed. You're guarding against `entries[0]` failing, but that code is in a closure that's only invoked if `entries` is nonempty.

Reply:
Ah. The plugin only responds with a run in the mapping if [the multiplexer retrieves at least 1 sampled step](https://github.com/tensorflow/tensorboard/blob/1c7b8ce22f94746d2892b3e599ceea226a3ab3dc/tensorboard/plugins/pr_curve/pr_curves_plugin.py#L158). Done.